core/indexer/processTransactions.go

```diff
@@ -280,6 +291,15 @@ func (tdp *txDatabaseProcessor) groupNormalTxsAndRewards(
 	return transactions, rewardsTxs, alteredAddresses
 }
 
+func (tdp *txDatabaseProcessor) shouldIndex(txHandler data.TransactionHandler) bool {
+	if !tdp.isInImportMode {
+		return true
+	}
+
+	destShardId := tdp.shardCoordinator.ComputeId(txHandler.GetRcvAddr())
+	return destShardId == tdp.shardCoordinator.SelfId()
```

Comment:
what about invalid transactions? they won't be indexed. 
also, a transaction could be `pending` for a long time due to small gas limit and validators avoiding it. I know it doesn't happen that often but it's something to think about

Reply:
Invalid transactions are detected and processed only in sender shard (for example: insufficient funds for transfer value but enough balance for the transaction fee)
Gas limit does not impact the transaction processing delay since the excess is returned to the sender. Gas price does. Since we are high throughput system, this is not of a major importance since, in the end, the destination shard will eventually process the transaction.